src/main/java/net/neoforged/neoforge/server/ServerLifecycleHooks.java

```diff
@@ -129,6 +133,18 @@ public static void handleExit(int retVal) {
         System.exit(retVal);
     }
 
+    private static <T> void stripKnownPack(boolean modified, Holder.Reference<T> holder, Registry<T> registry) {
+        if (modified) {
+            // If the object's networked data has been modified, we force it to sync by removing its original KnownPack info.
+            Optional<RegistrationInfo> originalInfo = registry.registrationInfo(holder.key());
+            originalInfo.ifPresent(info -> {
+                RegistrationInfo newInfo = new RegistrationInfo(Optional.empty(), info.lifecycle());
+                //noinspection unchecked
+                ((MappedRegistryAccessor<T>) registry).neoforge$getRegistrationInfos().put(holder.key(), newInfo);
+            });
+        }
+    }
+
```

Comment:
Would it not make more sense to not have the boolean `modified` property, and always execute the strip, as the name of the method would suggest?